src/System.ComponentModel.Composition.Registration/src/System/Threading/Lock.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    internal sealed class Lock : IDisposable
```

Comment:
@stephentoub as you are the expert for Threading, do you know if there's a way to not require these helper files (Lock, ReacLock, WriteLock)? I was able to get rid of all the MEF helper files but these remain...

Reply:
These structs aren't doing anything special: they're just making it so you can write code like:
```C#
using (new ReadLock(_lock))
{
    ...
}
```
instead of:
```C#
_lock.EnterReadLock();
try
{
    ...
}
finally { _lock.ExitReadLock(); }
```

Reply:
thanks